docs/spec.md

```diff
@@ -6285,6 +6336,33 @@ and [slicing](https://docs.python.org/3/reference/expressions.html#slicings)
 notations from Python are available to index into tensors, quantized tensors
 and tuples.
 
+* `convert_or_quantize_or_dequantize(x: Value, destination_type: Type) -> Value`
+is defined on tensors and returns the converted value of `x` based on the
+`type(x)` and `destination_type` as follows:
+
+```python
+def convert_or_quantize_or_dequantize(x: Value, destination_type: Type) -> Value:
```

Comment:
Agreed with your point. Thanks for the suggestions!
Let me propose picking  `to_destination_type`. To me it seems a good generic and concise choice without geting tied much into its implementation. `convert_any` looks fine too but I am biased towards the former. 

PTAL.

Reply:
I like generic _if there are use cases for being generic_. Rationale: Everything we support in the IR has implications for what backends must support.

With that said - Do we think it is trivial overhead for backends to support demotion if we are already supporting promotion? If so then probably OK to be generic. But if it would require more specialization for each of these ops, then perhaps we reconsider and restrict to use cases.

Reply:
Whoops, I misplaced this comment. Moving it to the other open thread and resolving this one.